Index: selinux-policy-2.8.4/src/agent/sysagent/boarddetectsysagent.cil
===================================================================
--- selinux-policy-2.8.4.orig/src/agent/sysagent/boarddetectsysagent.cil
+++ selinux-policy-2.8.4/src/agent/sysagent/boarddetectsysagent.cil
@@ -7,6 +7,11 @@
 
   (call conf.list_file_dirs (subj))
   (call conf.read_file_files (subj))
+  (call .uci.subj_type_transition (subj))
+  (call tmp.manage_file_dirs (subj))
+  (call tmp.manage_file_files (subj))
+  (call tmp.tmp_fs_type_transition_file (subj))
+  (call .tmp.deletename_fs_dirs (subj))
 
   (call .board.manage_file_files (subj))
 
@@ -43,6 +48,13 @@
 
   (call .ucode.subj_type_transition (subj))
 
+  (block tmp
+    (macro tmp_fs_type_transition_file ((type ARG1))
+      (call .tmp.fs_type_transition (ARG1 file file "*")))
+    (blockinherit .file.macro_template_dirs)
+    (blockinherit .file.macro_template_files)
+    (blockinherit .file.tmp.base_template))
+
   (block conf
 
     (filecon "/etc/board\.d" dir file_context)
Index: selinux-policy-2.8.4/src/agent/sysagent/hotplugsysagent.cil
===================================================================
--- selinux-policy-2.8.4.orig/src/agent/sysagent/hotplugsysagent.cil
+++ selinux-policy-2.8.4/src/agent/sysagent/hotplugsysagent.cil
@@ -18,6 +18,7 @@
     (call .block.read_sysfile_lnk_files (subj))
 
     (call .block.mount.subj_type_transition (subj))
+    (call .board.read_file_files (subj))
 
     (call .class.list_sysfile_dirs (subj))
     (call .class.read_sysfile_lnk_files (subj))
Index: selinux-policy-2.8.4/src/agent/sysagent/odhcpdsysagent.cil
===================================================================
--- selinux-policy-2.8.4.orig/src/agent/sysagent/odhcpdsysagent.cil
+++ selinux-policy-2.8.4/src/agent/sysagent/odhcpdsysagent.cil
@@ -26,6 +26,7 @@
     (call tmp.manage_file_dirs (subj))
     (call tmp.manage_file_files (subj))
     (call tmp.tmp_fs_type_transition_file (subj dir "odhcpd-piofolder"))
+    (call tmp.tmp_fs_type_transition_file (subj dir "odhcpd-piodir"))
     (call tmp.tmp_fs_type_transition_file (subj file ".odhcpd.leases"))
     (call tmp.tmp_fs_type_transition_file (subj file "odhcpd.leases"))
 
